### PR TITLE
Added ln funcationality to tensor trait, implemented logsoftmax, test of logsoftmax incomplete

### PR DESCRIPTION
**Pull request by Roee-87**
I added ln() functionality to tensors by updating all the necessary files to ensure that ln() can be implemented for fp, u32, i32.

I implemented logsoftmax functionality.

The test is incomplete.  Namely, I'm unsure how to represent negative floats in fixed point form.  The correct values for the logsoftmax output tensor are included in the error string of the assert statements in the logsoftmax_test.cairo file.

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ X] No

### CODE REVIEW

**Comment by credence0x** [comment]
Hi @Roee-87, great job!! I was checking out the repo and came across your PR. Wanted to try it out and I saw that you came across some issues that didn't make it compile.
I made a commit fixing all compilation issues here https://github.com/credence0x/orion/commit/ccd16bbf649110874aa53865934956714c03568a so please check it out. You can merge from that branch or copy and paste all changes to really get it if you don't already.

Feel me to chat on Discord if you have any other minor/major issues.

Other comments.
==> Positive FixedType: `Fixed::new(12, false)`
==> Negative FixedType: `Fixed::new(12, true)`
as seen here https://github.com/gizatechxyz/orion/blob/6f6b3101b93b182ef96810de1263e61b42cc0a80/src/numbers/fixed_point/core.cairo#LL3C1-L3C1
Basically, when `sign == true`, the value is `-mag` while when `sign == false` value is `+mag`. Sign there indicates whether the integer is signed or not.

File: src/tests/operators/nn/functional/logsoftmax/logsoftmax_test.cairo [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#[test]
+#[available_gas(20000000)]
+fn logsoftmax_test() {
+    let tensor = i32_tensor_2x2_helper();
+    let extra = ExtraParams { fixed_point: Option::Some(FixedImpl::FP8x23(())) };
+    let tensor = TensorTrait::new(tensor.shape, tensor.data, Option::Some(extra));
+
+    let mut result = NNTrait::logsoftmax(@tensor, 0).data;
+
+    // let test1 = *result.at(0).mag.into();
```
Comment (raphaelDkhn) [inline src/tests/operators/nn/functional/logsoftmax/logsoftmax_test.cairo:24]: Please remove comments

**Review by raphaelDkhn — COMMENTED** [review]
Thanks @Roee-87 Good first PR, Lgtm 👏 Just few things:

- Please add test for `tensor.ln()`
- Please run `scarb docgen` to generate doc from docstrings
- Please add your changes in `docs/CHANGELOG.md`
- Please add the new doc pages in `docs/SUMMARY.md`
- Please update `docs/apis/compatibility.md`

**Review by raphaelDkhn — APPROVED** [review]

**Comment by raphaelDkhn** [comment]
https://github.com/all-contributors please add @Roee-87 for code

**Comment by raphaelDkhn** [comment]
@all-contributors please add @Roee-87 for code
